darkdetect/_mac_detect.py

```diff
@@ -69,6 +80,38 @@ def isDark():
 def isLight():
     return theme() == 'Light'
 
-#def listener(callback: typing.Callable[[str], None]) -> None:
-def listener(callback):
-    raise NotImplementedError()
+
+def _listen_child():
+    """
+    Run by a child process, install an observer and print theme on change
+    """
+    class Observer(NSObject):
+        def callback_(self, _):
+            try:
+                print(theme(), flush=True)
+            except IOError:
+                os._exit(1)
+    observer = Observer.new()  # Keep a reference of the observer to keep it alive
+    NSDistributedNotificationCenter.defaultCenter().addObserver_selector_name_object_(
+        observer,  # Observer must be kept alive by a different reference
+        "callback:",
+        "AppleInterfaceThemeChangedNotification",
+        None
+    )
+    AppHelper.runConsoleEventLoop()
```

Comment:
Currently, this code does not work for me. If you want, I can provide another implementation based on `AppleInterfaceStyle` that works with the rest of the PR.

Reply:
If it doesn't work for you, then I'll try your alternative observer.

Reply:
Import: `from Foundation import NSObject, NSKeyValueObservingOptionNew, NSKeyValueChangeNewKey, NSUserDefaults`

```python
def _listen_child():
    OBSERVED_KEY = "AppleInterfaceStyle"

    class Observer(NSObject):
        def observeValueForKeyPath_ofObject_change_context_(
            self, path, object, changeDescription, context
        ):
            result = changeDescription[NSKeyValueChangeNewKey]
            if result is None:
                result = "Light"
            print(f'{result}', flush=True)

    observer = Observer.new()
    defaults = NSUserDefaults.standardUserDefaults()

    defaults.addObserver_forKeyPath_options_context_(
        observer, OBSERVED_KEY, NSKeyValueObservingOptionNew, 0
    )

    AppHelper.runConsoleEventLoop(installInterrupt=True) 
```

Reply:
I'll give it a shot.